client/CS130_Team13/Assets/Scripts/RemoteController.cs

```diff
@@ -7,75 +7,129 @@
+    }
+
+    public override string ToString() {
+        return "name: " + name + ", id: " + id;
+    }
 }
 
-public class RemoteController : MonoBehaviour
-{
-  private QSocket socket;
-  private UserInfo userInfo;
-  private UserInfo opponentInfo;
-
-  void Start()
-  {
-    StartCoroutine(InitializeGame());
-  }
-
-  IEnumerator InitializeGame()
-  {
-    int id;
-    using (UnityWebRequest webRequest = UnityWebRequest.Get("http://localhost:3000/user_id"))
-    { // TODO: ERROR CHECK
-      // Request and wait for the desired page.
-      yield return webRequest.SendWebRequest();
-      id = System.Convert.ToInt32(webRequest.downloadHandler.text);
-      Debug.Log(webRequest.downloadHandler.text);
+public class RemoteController : MonoBehaviour {
+    private QSocket socket;
+    private UserInfo userInfo;
+    private UserInfo opponentInfo;
+
+    void Start() {
+        StartCoroutine(InitializeGame());
+    }
+
+    IEnumerator InitializeGame() {
+        int id;
+        using (UnityWebRequest webRequest = UnityWebRequest.Get("http://localhost:3000/user_id")) {
+            // TODO: ERROR CHECK
+            // Request and wait for the desired page.
+            yield return webRequest.SendWebRequest();
+            id = System.Convert.ToInt32(webRequest.downloadHandler.text);
+            Debug.Log(webRequest.downloadHandler.text);
+        }
+        // set User Info
+        userInfo = new UserInfo("pickles", id);
+        setupSocket();
+    }
+
+    private void setupSocket() {
+        socket = IO.Socket("http://localhost:3000");
+
+        // this happens on connect
+        socket.On(QSocket.EVENT_CONNECT, () => {
+            Debug.Log("Connected");
+            // sends server the user info
+            socket.Emit("hello", JsonUtility.ToJson(userInfo));
+        });
+
+        // this happens when server emits to event called initiate
+        socket.On("initiate", (data) => {
+            Debug.Log(data);
+        });
+
+        socket.On("gameplay", (data) => {
+            // TODO switch to socket.id use
+            Debug.Log(data);
+            opponentInfo = JsonUtility.FromJson<UserInfo>(data.ToString());
+            Debug.Log(opponentInfo.ToString());
+        });
+    }
+
+    private void Destroy() {
+        Debug.Log("byeee");
+        socket.Disconnect();
+    }
+
+    // --------------------------------------------------------------------------------
+    // The following functions should not do anything other than call other functions
+    // present elsewhere in the code. They exist as an abstraction layer between
+    // server and client: if the implementation of one is changed, then you'll have
+    // to change some function names here -- but the benefit is that you _won't_ have
+    // to go digging through the rest of the code to find all the references in the
+    // other half of the project.
+    // --------------------------------------------------------------------------------
+
+    /// <summary>
```

Comment:
maybe it can automatically match when this object is created? so this isnt needed

Reply:
done